graylog2-web-interface/src/components/common/Wizard.jsx

```diff
@@ -0,0 +1,109 @@
+
+import WizardStyle from './Wizard.css';
+
+/**
+ * Component that renders a wizard, letting the consumers of the component
+ * manage the state. It will render at least two columns: First column will hold
+ * the steps the wizard will take. Second column will render the component of the
+ * selected step. In a optional third column the consumer can render a preview.
+ */
+class Wizard extends React.Component {
+  static propTypes = {
+    /**
+     * Array of objects which will describe the wizard. The object must
+     * contain a unique 'key' attribute, a 'title' which will be shown as step link on the left side and
+     * a 'component' attribute which will hold the component which is to render for the step.
+     *
+     * e.g:
+     * [\[<br />
+     * &nbsp;&nbsp;{key: 'key1', title: 'General Information', component: (&lt;Acomponent1 /&gt;)},<br />
+     * &nbsp;&nbsp;{key: 'key2', title: 'Details', component: (&lt;Acomponent2 /&gt;)},<br />
+     * &nbsp;&nbsp;{key: 'key3', title: 'Preview', component: (&lt;Acomponent3 /&gt;)},<br />
+     * \]]
+     */
+    steps: PropTypes.arrayOf(PropTypes.object).isRequired,
+    /**
+     * Callback which is called when the user changes the step. As an argument the callback gets the key
+     * of the next step.
+     */
+    onStepChange: PropTypes.func,
+    /** Optional component which can be rendered on the right side e.g a preview */
+    children: PropTypes.element,
+  };
+
+  static defaultProps = {
+    children: undefined,
+    onStepChange: () => {},
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      selectedStep: props.steps[0].key,
+    };
+
+    this._wizardChanged = this._wizardChanged.bind(this);
+    this._onNext = this._onNext.bind(this);
+    this._onPrevious = this._onPrevious.bind(this);
+  }
+
+  _wizardChanged(eventKey) {
+    this.props.onStepChange(eventKey);
+    this.setState({ selectedStep: eventKey });
+  }
+
+  _disableButton(direction) {
+    const len = this.props.steps.length;
+    const position = direction === 'next' ? (len - 1) : 0;
+    return this.props.steps[position].key === this.state.selectedStep;
+  }
+
+  _onNext() {
+    this._wizardChanged(this.props.steps[this._getSelectedIndex() + 1].key);
+  }
+
+  _onPrevious() {
+    this._wizardChanged(this.props.steps[this._getSelectedIndex() - 1].key);
+  }
+
+  _getSelectedIndex() {
+    return this.props.steps.map(step => step.key).indexOf(this.state.selectedStep);
+  }
+
+  render() {
+    return (
+      <Row className="content">
+        <Col md={2} className={WizardStyle.subnavigation}>
+          <Nav stacked bsStyle="pills" activeKey={this.state.selectedStep} onSelect={this._wizardChanged}>
+            {this.props.steps.map((navItem) => {
+              return (<NavItem key={navItem.key} eventKey={navItem.key}>{navItem.title}</NavItem>);
+            })}
+          </Nav>
+          <br />
+          <Row>
+            <Col md={6}>
```

Comment:
Small issue: On an [`md` screen size](https://getbootstrap.com/docs/3.3/css/#grid-options), the previous and next buttons are overlapped. I think we could make both buttons smaller, i.e. `bsSize="small"`, avoiding the overlapping and still looking alright.

Normal buttons:
<img width="938" alt="screen shot 2018-04-02 at 14 04 14" src="https://user-images.githubusercontent.com/716185/38195656-d689f286-367e-11e8-85ca-0b31b65560d3.png">

Small buttons:
<img width="941" alt="screen shot 2018-04-02 at 14 04 51" src="https://user-images.githubusercontent.com/716185/38195663-dcf57762-367e-11e8-86bb-8f3d73de63d6.png">

Reply:
We should also use `sm` or `xs` to set the grid size for these buttons instead of `md`. In that way both buttons will keep their grid layout even if the screen is a bit smaller, instead of expanding into different rows. We usually don't care too much about how things look on small screens, but this one is easy to solve and I thought it was worth mentioning.